### PR TITLE
Extended controller for new localization TRNG-1414

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Added the new localization loading mechanics to the existing controller.

only works with: https://github.com/Innoactive/Creator/pull/264

### CODE REVIEW

File: Runtime/CourseController/Controllers/ILocalizationProvider.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using Innoactive.Creator.Core.Internationalization;
+
+namespace Innoactive.Creator.UX
+{
+    public interface ILocalizationProvider
```
Comment (tomwim) [inline Runtime/CourseController/Controllers/ILocalizationProvider.cs:5]: Short xml doc would be nice here

File: Runtime/CourseController/Controllers/BaseCourseController.cs [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
+using Innoactive.Creator.Core.Internationalization;
 using UnityEngine;
 
 namespace Innoactive.Creator.UX
 {
     /// <summary>
     /// Base course controller which instantiates a defined prefab.
     /// </summary>
-    public abstract class BaseCourseController : ICourseController
+    public abstract class BaseCourseController : ICourseController, ILocalizationProvider
```
Comment (tomwim) [inline Runtime/CourseController/Controllers/BaseCourseController.cs:11]: I have the feeling we are creating a monster with the CourseController :D but it probably needs a full rework at some point then

**Review by tomwim — APPROVED** [review]
Approved, just add the XML doc please. Thanks!

File: Runtime/CourseController/Controllers/BaseCourseController.cs (hunk 2 above)
Comment (SimonTheSourcerer) [reply Runtime/CourseController/Controllers/BaseCourseController.cs:11]: Naaa... course controller also decides which mode is used and depending on the mode it should also decided which language configuration is loaded.

**Review by tomwim — APPROVED** [review]

**Pull request merged**